Vendor/filesystem.cmake

```diff
@@ -0,0 +1,27 @@
+include(FetchContent)
+include(CheckCXXSourceCompiles)
+
+check_cxx_source_compiles("
+#ifdef __APPLE__
+#include <Availability.h> // for deployment target to support pre-catalina targets without std::fs
+#endif
+#if ((defined(_MSVC_LANG) && _MSVC_LANG >= 201703L) || (defined(__cplusplus) && __cplusplus >= 201703L)) && defined(__has_include)
+#if __has_include(<filesystem>) && (!defined(__MAC_OS_X_VERSION_MIN_REQUIRED) || __MAC_OS_X_VERSION_MIN_REQUIRED >= 101500)
+#define GHC_USE_STD_FS
+#endif
+#endif
+#ifndef GHC_USE_STD_FS
+#error \"std::filesystem missing or unusable\"
+#endif
```

Comment:
Can you change this to:

1. *NEVER* use `ghc::filesystem` on Windows; the UTF-8 changes they've made means that it will be a big breaking change there, as the 8-bit locale is practically never UTF-8, compared to always being UTF-8 on MacOS. If C++20's <filesystem> is not available on Windows, the build/configure must fail.
2. see if just using __has_include + the __lib_cpp feature macro are good enough

There's no worry of similar version problems with Windows due to these plugins using https://github.com/microsoft/WindowsAppSDK/blob/main/docs/Coding-Guidelines/HybridCRT.md - from a *nix point of view, that's similar (but not quite the same) as statically linking libc but dynamically linking libc++/libstdc++

Reply:
I took the check directly out of `ghc_filesystem`'s [README](https://github.com/gulrak/filesystem/#using-it-as-forwarding-implementation-header). I assume they've carefully tuned it to only apply on the systems where there is a non-working `std::filesystem` so I didn't want to mess with the logic. It's definitely not as simple as checking for the presence of the include file; the include exists and only causes compilation errors if you try to instantiate a `path` object.

I'm not sure what you mean by "the UTF-8 changes they've made". It looks like the default behavior of `ghc_filesystem` on Windows is to use `std::wstring` as storage. You can set the define `GHC_WIN_DISABLE_WSTRING_STORAGE_TYPE` to make it use utf8-encoded `std::string` instead. Surely one of those matches the modern Windows implementation, right? I would much rather make it work on old Windows than just break it.

Reply:
>  It's definitely not as simple as checking for the presence of the include file; the include exists and only causes compilation errors if you try to instantiate a path object.

Does checking for `__cpp_lib_filesystem` after including `<filesystem>` prevent this?

> I'm not sure what you mean by "the UTF-8 changes they've made".

https://github.com/gulrak/filesystem#differences-in-api

While `std::filesystem::path` is usually backed by `wstring` on Windows, the conversion from `std::string` or `char*` is locale-sensitive, but with `ghc::filesystem` it is not

Reply:
> > It's definitely not as simple as checking for the presence of the include file; the include exists and only causes compilation errors if you try to instantiate a path object.
> 
> Does checking for `__cpp_lib_filesystem` after including `<filesystem>` prevent this?

It does not; that macro is always defined.

> > I'm not sure what you mean by "the UTF-8 changes they've made".
> 
> https://github.com/gulrak/filesystem#differences-in-api
> 
> While `std::filesystem::path` is usually backed by `wstring` on Windows, the conversion from `std::string` or `char*` is locale-sensitive, but with `ghc::filesystem` it is not

Hm, okay. I think this is only a problem in practice if someone is taking a `path` and giving it to a pre-`*W` Windows API. Nevertheless, since I'm not personally affected I'm happy to leave fixing this for the next person who encounters it. Changed.

Reply:
>  if someone is taking a path and giving it to a pre-*W Windows API

Both directions - and Elgato's own demo plugins use the A versions, e.g. https://github.com/elgatosf/streamdeck-cpu/blob/e5a04d3d0cc22cabe11221104909a3e0ccb6292f/Sources/Common/ESDUtilitiesWindows.cpp#L234 ; given this was the first release of this project that supports `std::filesystem::path`, mixing the two needs to work correctly

> I'm happy to leave fixing this for the next person who encounters it.

Shouldn't be a problem on Windows, as compiler and C++ library features are independent of OS version.